File: simpleMediaPlayer/app/src/main/java/com/daeseong/simplemediaplayer/MusicApplication.java
package com.daeseong.simplemediaplayer;

import android.app.Application;
import android.content.Context;
import android.content.res.Configuration;

public class MusicApplication extends Application {

    private static final String TAG = MusicApplication.class.getSimpleName();

    private static MusicApplication mInstance;
    public static synchronized MusicApplication getInstance() {
        return mInstance;
    }

    private static Context mContext;
    public static Context getAppContext(){
        return mContext;
    }

    @Override
    public void onCreate() {
        super.onCreate();

        mInstance = this;
        mContext = getApplicationContext();
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
    }

}

File: simpleMediaPlayer(kotlin)/app/src/main/java/com/daeseong/simplemediaplayer/MusicApplication.kt
package com.daeseong.simplemediaplayer

import android.app.Application
import android.content.Context
import android.content.res.Configuration

class MusicApplication : Application() {

    private val tag: String = MusicApplication::class.java.simpleName

    companion object {
        private lateinit var mContext: Context
        private lateinit var mInstance: MusicApplication

        fun getContext(): Context {
            return mContext.applicationContext
        }

        fun getInstance(): MusicApplication {
            return mInstance
        }
    }

    override fun onCreate() {
        super.onCreate()

        mContext = this
        mInstance = this
    }

    override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)
    }

}